### PR TITLE
Cps/ Refactoring define wake process

### PR DESCRIPTION
**Pull request by marcnunezc**
Hi,

This is hopefully the last PR before merging #4642 which will only contain the element implementation and tests. 

The changes in this PR were implemented by @inigolcanalejo  in #4642 after @rubenzorrilla suggestions. 

We read the wake_direction from the direction of the free_stream_velocity, which allows to avoid defining it in the settings. Also private methods are renamed with "__".

Thanks!

### CODE REVIEW

File: applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py
Comment (rubenzorrilla) [inline applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py:52]: Why name mangling in all these methods?

File: applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py
Comment (rubenzorrilla) [inline applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py]:
```suggestion
    def _SetWakeDirectionAndNormal(self):
```
You are not reading it.

File: applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py
Comment (rubenzorrilla) [inline applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py]: It would be nice to name this member variable with a more self-explicative name.

File: applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py
Comment (rubenzorrilla) [inline applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py]: Why are you avoiding the model part API? I'd suggest to use either the `AddElement` or `AddElements` methods. Note that the second one is more efficient but would require to create an auxiliary array containing the IDs of the elements to be added.

File: applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py
Comment (rubenzorrilla) [inline applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py:174]: Which is the purpose of this decorator? I've never used it.

File: applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py
Comment (rubenzorrilla) [inline applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py:213]: Note that in here you are retrieving the discontinuous elemental distances, which might not coincide with the nodal ones if you call this method after using the modify distance process.

File: applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py
Comment (marcnunezc) [reply applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py:52]: This is intended to follow the style guide, which was pointed out in:
https://github.com/KratosMultiphysics/Kratos/pull/4642#discussion_r274928255

File: applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py
Comment (marcnunezc) [reply applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py]: Ok I agree!

File: applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py
Comment (marcnunezc) [reply applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py]: Ok will try to use the modelpart API!

File: applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py
Comment (marcnunezc) [reply applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py:213]: In this specific case we are not using the modified_distance process since we are analytically computing the distances here (note that is the distance function that we use for the wake, which is different from the distance funcion that we will use for the embedded case):
https://github.com/KratosMultiphysics/Kratos/blob/529adfa67c2859e68711846eb6dd977374822eb9/applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py#L146

File: applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py
Comment (marcnunezc) [reply applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py:174]: To define the method as static, meaning that it cannot access the `self` operator, so it can't read/modify any member variable or method from the class.

File: applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py
Comment (rubenzorrilla) [reply applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py:52]: The point is that in Python doing __MyMethod mangles the name, which is not private at all... In any case I wouldn't block the PR because of this.

File: applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py
Comment (rubenzorrilla) [reply applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py:174]: Good to know! :+1:

File: applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py
Comment (rubenzorrilla) [reply applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py:213]: OK. But how are you computing the distances? By means of the `CalculateDistanceToSkinProcess` process? If this is the case, the nodal values do not strictly coincide with the elemental ones.

File: applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py
Comment (philbucher) [reply applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py:52]: This was the agreement for "private" methods, which was discussed long ago
Ofc you could still access them but it gives a hint to the user to not do it...

Comment (rubenzorrilla) [reply applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py:52]: OK then! From now on I'll follow this convention.

File: applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py
Comment (marcnunezc) [reply applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py:213]: No, we are not using the CalculateDistanceToSkinProcess!
What I meant is that we are using a custom ComputeDistanceToWake method (this is 2D).

In any case, thanks for pointing it out because for the 3D we will have to use the CalculateDistanceToSkinProcess and your point may be important then.

File: applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py
Comment (marcnunezc) [reply applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py]: Yes you are right` self.te `is not clear enough,

File: applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py
Comment (rubenzorrilla) [reply applications/CompressiblePotentialFlowApplication/python_scripts/define_wake_process_2d.py:213]: Then I suggest you to leave a comment in the current version as a reminder.

**Comment by marcnunezc** [comment]
@rubenzorrilla if you have no further comments this should be ready to merge! After finishing #4642 I would adapt this process to c++

**Review by rubenzorrilla — APPROVED** [review]
Please go ahead.

**Pull request merged**
